etna/models/nn/chronos/base.py

```diff
@@ -0,0 +1,277 @@
+import os
+import warnings
+import zipfile
+from abc import abstractmethod
+from pathlib import Path
+from typing import Dict
+from typing import List
+from typing import Sequence
+from typing import Union
+from urllib import request
+
+import pandas as pd
+
+from etna import SETTINGS
+from etna.datasets import TSDataset
+from etna.distributions import BaseDistribution
+from etna.libs.chronos import BaseChronosPipeline
+from etna.libs.chronos import ChronosBoltModelForForecasting
+from etna.libs.chronos import ChronosModelForForecasting
+from etna.models.base import PredictionIntervalContextRequiredAbstractModel
+
+if SETTINGS.chronos_required:
+    import torch
+
+
+class ChronosBaseModel(PredictionIntervalContextRequiredAbstractModel):
+    """Base class for Chronos-like pretrained models."""
+
+    def __init__(
+        self,
+        path_or_url: str,
+        encoder_length: int,
+        device: str,
+        dtype: torch.dtype,
+        cache_dir: Path,
+    ):
+        """
+        Init Chronos-like model.
+
+        Parameters
+        ----------
+        path_or_url:
+            Path to the model. It can be huggingface repository, local path or external url
+            - If huggingface repository, see available models in ``list_models`` of appropriate model class.
+            During the first initialization model is downloaded from huggingface and saved to local ``cache_dir``.
+            All following initializations model will be loaded from ``cache_dir``. See ``pretrained_model_name_or_path`` parameter of :py:func:`transformers.PreTrainedModel.from_pretrained`
+            - If local path, model will not be saved to local ``cache_dir``.
+            - If external url, it must be zip archive with the same name as model directory inside. Model will be downloaded to ``cache_dir``.
```

Comment:
Are we checking it somehow?

Reply:
No. I think real check should be complex like checking `***.safetensor` or `pytorch_model.bin` or `config.json` is in directory. As the main purpose is developer cases, so we can pay less attention to it.